### DISCUSSION TITLE
Explanation

### DISCUSSION THREAD

**Discussion by vitezprchal** [a-h/templ#815] [Q&A]
Hi,

I've reviewed the documentation, but I'm having difficulty understanding this implementation:
[main.go in the integration-gin example](https://github.com/a-h/templ/blob/main/examples/integration-gin/main.go).

Could someone please provide further clarification on what "HTMLRender" does in this context?

**Comment by joerdav** [comment]
Hey, it allows you to pass a templ component directly to the gin `Render` component.

`c.Render(http.StatusOK, r)`
